subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -109,6 +109,11 @@ struct tls_context {
 
 		/** DTLS role, client by default. */
 		int8_t role;
+
+		/** NULL-terminated list of allowed application layer
+		 * protocols.
+		 */
+		const char *alpn_list[CONFIG_NET_SOCKETS_TLS_MAX_APP_PROTOCOLS + 1];
```

Comment:
We probably do not need this list if ALPN is not compiled in. Perhaps provide zero length array in that case so that we do not need to put ifdefs all over the place.